CleverGirl/CleverGirl/Helper/AOHelper.cs

```diff
@@ -457,7 +489,16 @@ private static bool DoesMechFavorMelee(Mech attacker)
                     }
                     else
                     {
-                        rawRangedDam += (int)(weapon.DamagePerShot * weapon.ShotsWhenFired);
+                        int optimalDamage = 0;
+                        AmmoModePair currentAmmoMode = weapon.getCurrentAmmoMode();
+                        foreach (AmmoModePair ammoModePair in weapon.getAvaibleFiringMethods())
+                        {
+                            weapon.ApplyAmmoMode(ammoModePair);
+                            optimalDamage = Mathf.Max(optimalDamage, (int)(weapon.DamagePerShot * weapon.ShotsWhenFired));
```

Comment:
Why are you doing a floating point compare here, when you've cast to int and optimalDamage is 0?

Reply:
Additionally, this calc isn't taking quite a bit of CAC logic into account. It's ignoring AoE fragmentation, AoE falloff, etc. CG probably ignored that as well, so more just a caution since I'm not entirely sure what this block of checking is intended to achieve.

Reply:
It's a very naive approach to trying to find if the mech should consider melee attacks. I did not change it from existing code besides adding ammomodes. 
Adding a todo to look at using the CAC logic, but there's a performance impact of that I would imagine.